package.json

```diff
@@ -158,10 +158,27 @@
     "uuid": "^9.0.1"
   },
   "size-limit": [
+    {
+      "brotli": false,
+      "path": "./packages/teams-js/dist/esm/packages/teams-js/src/index.js",
+      "import": "{ app, authentication, pages }",
+      "limit": "56.24 KB"
+    },
     {
       "brotli": false,
       "path": "./packages/teams-js/dist/esm/packages/teams-js/src/index.js",
       "import": "*"
     }
+  ],
+  "size-limit-explanation": [
+    "There are 2 configurations for size-limit. The first configuration is used to test treeshakability in teams-js. If treeshaking is broken in any capability other than `app`, `authentication`, or `pages`, then the size of the build",
+    "will increase above the set 58.033KB. The limit is currently set to the exact size of the 3 specified files when built. If your build ever fails because of this size-limit check there are a few scenarios you may fall into:",
```

Comment:
Is this a strict limit or do we allow for very minor size changes like <1% as an example?

Reply:
Since the point of the limit is to detect treeshaking breakages, I wanted to have it fairly close to the actual size of the 3 files listed, but if you think it would be less of a headache for developers, I'm open to increasing the limit slightly to allow for a bit of wiggle room. I just want to be cautious about creating a habit of permissiveness when it comes to increasing the size limit as I feel it could lose some of its effectiveness.

Reply:
I think my question is moreso can we guarantee that these 3 files will always build to that exact size if no changes are made?

Additionally, this is an upper limit? What happens if we make major strides to lower the size of these files is there any suggestions for when we should lower this upper limit in the future?

Reply:
So as long as no changes are made to these 3 files, or any of their dependencies the size won't change. I went ahead and bumped up the size limit slightly, but it's really hard to know a good 'buffer' amount without just putting it into practice and seeing. If the size goes down, we will need to adjust the limit to re-correspond with the new size.